WordPress/gutenberg
# Block editor: add a keyboard shortcut to create group from the selected blocks

Closes #42227

## What?
This PR adds a keyboard shortcut to create a group block from the selected blocks.

## Why?
Many design tools provide shortcuts for grouping elements. It would make sense to add a similar shortcut, especially since the grouping of blocks should be done frequently in the site editing.

## How?
To explore which key combinations were available, I tested all variations that included the letter `G`. The list can be found in [this comment](https://github.com/WordPress/gutenberg/issues/42227#issuecomment-1374842421).

The results show that the following two modifiers are available:

- `primaryAlt`: `⌘ command` + `⌥ option` + `G` on Mac / `ctrl` + `alt` + `G` on Window
- `secondary`: `⌘ command` + `shift` + `⌥ option` + `G` on Mac / `ctrl` + `shift` + `alt` + `G`

~I chose `primaryAlt` because the `secondary` modifier could be used for ungrouping (unwrapping) in the future.~

~However, as noted in this comment, it appears that Ctrl + Alt is usually used for Windows OS-level functions. I would appreciate your advice on whether it makes sense to add this shortcut and whether it will cause any confusion to the user.~

Update: After some discussion, we adopted the simpler `⌘ command + G` (`ctrl + G`) as a shortcut.

Also, this shortcut will not allow to create a group block if a single block is selected. This is consistent with the behavior of group block transformation.

## Testing Instructions

- Insert multiple blocks and select them all.
- On Windows, press `ctrl` + `alt` + `G`. On Mac, press `⌘ command` + `⌥ option` + `G`.
- Confirm that the selected block is wrapped in a group block.
- The second shortcut should not be executed because the group block is selected.

## Screenshots or screencast

https://user-images.githubusercontent.com/54422211/211314612-b1399643-c770-4774-a323-ebb95e0c2fc4.mp4
